### PR TITLE
feature: Get started page, create customer and create new sale

### PR DESCRIPTION
**Pull request by cheedinmar**
work in progress...
not to merge

### CODE REVIEW

File: src/components/Search.vue [hunk 1]
```diff
@@ -10,11 +10,11 @@
                     <div class="w-full flex-1 border-r border-gray-200 py-2 md:py-4">
                         <label class="text-gray-400 font-semibold text-xs px-4">SEARCHING FOR</label>
                         <input type="text" name="phone" id="phone" class="block w-full rounded-md border-0 focus:ring-0 sm:text-sm px-4"
-                            placeholder="08010000000" aria-describedby="email-description" />
+                            placeholder="08010000000" aria-describedby="email-description" v-model="phoneNumber" @input="Search" />
                     </div>
                     <div class="flex justify-center items-center py-2 md:py-4 px-4">
                         <button type="button"
-                            class="inline-flex items-center rounded-full border border-transparent bg-indigo-600 px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-0">
+                            class="inline-flex items-center rounded-full border border-transparent bg-primary px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-0">
```
Comment (AltaraNg) [inline src/components/Search.vue:17]: if the background is `bg-primary`, can we change the hover state too?

File: src/components/Search.vue [hunk 2]
```diff
@@ -25,6 +25,21 @@
     </div>
 </template>
 
-<script setup>
```
Comment (AltaraNg) [inline src/components/Search.vue:28]: Let's revert back to composition api, that's the convention we are using now

File: src/services/ApiService.js [hunk 3]
```diff
@@ -2,7 +2,8 @@ import axios from "axios";
 import { handleError } from "../utilities/GlobalFunctions";
 export class Apiservice {
   constructor() {
-    this.baseUrl = process.env.VUE_APP_API_URL;
+    this.baseUrl = "https://bnpl-play-api.herokuapp.com/api/";
```
Comment (AltaraNg) [inline src/services/ApiService.js:5]: can you revert this?

File: src/views/GetStarted.vue [hunk 4]
```diff
@@ -67,38 +35,388 @@ const state = reactive({
...
+                            </div>
+                        </SideModal>
                     </div>
                 </div>
             </div>
         </div>
     </div>
 </template>
-
+<script>
```
Comment (AltaraNg) [inline src/views/GetStarted.vue:201]: Can you revert back to composition api?

File: src/views/SignUp.vue [hunk 5]
```diff
@@ -2,14 +2,14 @@
   <div class="relative bg-white">
     <div class="lg:absolute lg:inset-0">
       <div class="lg:absolute lg:inset-y-0 lg:right-0 lg:w-1/2">
-        <img class="h-56 w-full object-cover lg:absolute lg:h-full" src="https://images.unsplash.com/photo-1556761175-4b46a572b786?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1567&q=80" alt="" />
+        <img class="h-56 w-full object-cover lg:absolute lg:h-full opacity-50" src="../assets/images/market.jpg" alt="" />
```
Comment (AltaraNg) [inline src/views/SignUp.vue:5]: use the `@` alias here, so we don't run into path issue

File: src/views/SignUp.vue [hunk 6]
```diff
@@ -45,55 +45,110 @@
...
             </div>
           </form>
         </div>
       </div>
     </div>
   </div>
-</template> 
\ No newline at end of file
+</template> 
+<script>
```
Comment (AltaraNg) [inline src/views/SignUp.vue:76]: Use composition api

File: src/store/action.js [hunk 7]
```diff
@@ -1,36 +1,77 @@
...
+                    verification_id: response.result.credit_check_verification.id,
+                    verification_status: response.result.credit_check_verification.status,
+                },
+            });
+            handleSuccess("Success");
+        } 
+    });
 };
+
+//  setInterval(() => {
```
Comment (teebarg) [inline src/store/action.js:68]: Can we remove this if it'd no longer need?

File: src/views/CreateOrder.vue [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+                x.down_payment_rate_id === payment_type_id.value?.id &&
+                x.repayment_duration_id === Order.repayment_duration_id?.id
+            );
+        });
+        const { total, actualDownpayment, rePayment } = calculate(Order.amount, Data, params, 0);
+
+        OrderResult.value.total = total;
+        OrderResult.value.actualDownpayment = actualDownpayment;
+        OrderResult.value.rePayment = rePayment;
+        // console.log(OrderResult.total, OrderResult.actualDownpayment, OrderResult.rePayment);
```
Comment (teebarg) [inline src/views/CreateOrder.vue:232]: Remove this

File: src/views/CreateOrder.vue [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+    } catch (e) {
+        console.log(e);
```
Comment (teebarg) [inline src/views/CreateOrder.vue:234]: Remove this

File: src/views/GetStarted.vue [hunk 10]
```diff
@@ -1,104 +1,148 @@
...
+     phone_number.value = phoneNumber;
+    
+    FindCustomer();
+}
+const FindCustomer = () => {
+    FilteredCustomer.value = Customers.value.filter((customer) => {
+        return customer.telephone.includes(phone_number.value);
+    });
+    return phone_number.value ? FilteredCustomer.value : Customers.value?.slice(0, 3);
+    // console.log( FilteredCustomer, phone_number.value)
```
Comment (teebarg) [inline src/views/GetStarted.vue:136]: remove this

File: src/services/ApiService.js [hunk 11]
```diff
@@ -22,16 +26,20 @@ export class Apiservice {
...
+    store.dispatch("loader/show", { root: true });
     try {
       let result = await this.api_connector.get(url, this.requestConfig);
       this.resetRequestConfig();
+      store.dispatch("loader/hide", { root: true });
       return result;
     } catch (error) {
-      this.handleErrors(error);
+      store.dispatch("loader/hide", { root: true });
+      handleError(error.response.data.message);
```
Comment (teebarg) [inline src/services/ApiService.js:37]: Use optional chaining

File: src/store/state.js [hunk 12]
```diff
@@ -1,7 +1,192 @@
...
-    loading: false,
-    error: null,
-  },
+    userdata: {},
+    status: {
+        loading: false,
+        error: null,
+    },
+    Customers: [
+        {
```
Comment (teebarg) [inline src/store/state.js:8]: Do we still need this since we are integrating APIs now?

**Review by AltaraNg — APPROVED** [review]

**Pull request merged**